src/java/org/apache/cassandra/db/virtual/SettingsTable.java

```diff
@@ -59,102 +56,83 @@ final class SettingsTable extends AbstractVirtualTable
                            .addPartitionKeyColumn(NAME, UTF8Type.instance)
                            .addRegularColumn(VALUE, UTF8Type.instance)
                            .build());
-        this.config = config;
+        registerConverters();
+        this.configurationRegistry = configurationRegistry;
     }
 
-    @Override
-    public DataSet data(DecoratedKey partitionKey)
+    private void registerConverters()
     {
-        SimpleDataSet result = new SimpleDataSet(metadata());
-        String name = UTF8Type.instance.compose(partitionKey.getKey());
-        if (BACKWARDS_COMPATABLE_NAMES.containsKey(name))
-            ClientWarn.instance.warn("key '" + name + "' is deprecated; should switch to '" + BACKWARDS_COMPATABLE_NAMES.get(name) + "'");
-        if (PROPERTIES.containsKey(name))
-            result.row(name).column(VALUE, getValue(PROPERTIES.get(name)));
-        return result;
+        converterRegistry.put(Boolean.class, CassandraRelevantProperties.BOOLEAN_CONVERTER);
+        converterRegistry.put(boolean.class, CassandraRelevantProperties.BOOLEAN_CONVERTER);
+        converterRegistry.put(Integer.class, CassandraRelevantProperties.INTEGER_CONVERTER);
+        converterRegistry.put(int.class, CassandraRelevantProperties.INTEGER_CONVERTER);
+        converterRegistry.put(DurationSpec.LongMillisecondsBound.class,
+                              (PropertyConverter<DurationSpec.LongMillisecondsBound>) value -> DurationSpec.from(value, DurationSpec.LongMillisecondsBound.class));
     }
 
-    @Override
-    public DataSet data()
+    /**
+     * Setter for the property.
+     * @param name the name of the property.
+     * @param value the string representation of the value of the property to set.
+     */
+    private void setProperty(String name, String value)
     {
-        SimpleDataSet result = new SimpleDataSet(metadata());
-        for (Map.Entry<String, Property> e : PROPERTIES.entrySet())
-            result.row(e.getKey()).column(VALUE, getValue(e.getValue()));
-        return result;
+        Class<?> setterType = configurationRegistry.getPropertyType(name);
+        PropertyConverter<?> converter = converterRegistry.get(setterType);
+        if (converter == null)
+            throw invalidRequest("Unknown converter for property with name '%s' and type '%s'", name, setterType);
+
+        configurationRegistry.update(name, value == null ? null : converter.convert(value));
```

Comment:
I don't follow why this is needed.  You have a map of "real" type to "PropertyConverter", so can't you just do

```
Property p = PROPERTIES.get(name);
if (p == null) throw unknownProperty(name);
p.set(DatabaseDescriptor.getRawConfig(), getConverter(p.getType()).convert(value));
```

If there is a missing converter that means we have a bug, we don't support all types we support...